GliaWidgets/ViewModel/Chat/ChatViewModel.swift

```diff
@@ -308,12 +309,20 @@ extension ChatViewModel {
 // MARK: History
 
 extension ChatViewModel {
-    private func loadHistory() {
-        let messages = chatStorageState().messages(interactor.queueID)
-        let items = messages.compactMap { ChatItem(with: $0, fromHistory: environment.loadChatMessagesFromHistory()) }
-        historySection.set(items)
-        action?(.refreshSection(historySection.index))
-        action?(.scrollToBottom(animated: false))
+    private func loadHistory(_ completion: @escaping () -> Void) {
+        environment.fetchChatHistory { [weak self] result in
+            defer { completion() }
```

Comment:
yes, it will be executed after `guard self` also that is not a good point since if we got `self` is nil it is some kind of unexpected and very strange case, so I believe it better to do nothing than execute completion. I'll update the code and move to defer after guard.

Reply:
And what happens with the code that expects a completion to be fired? It would wait forever? That is why I was thinking that maybe it would be better to fire a completion with a result, then it would be failure for the `else` in the `guard`, and success for the rest